src/map-extent.js

```diff
@@ -48,58 +62,724 @@ export class MapExtent extends HTMLElement {
+                boundsFallback.zoom,
+                projection,
+                M.axisToCS(axis)
+              );
+            inp.setAttribute('min', axisBounds.min[M.axisToXY(axis)]);
+            inp.setAttribute('max', axisBounds.max[M.axisToXY(axis)]);
+          }
+
+          inputs.push(inp);
+          includesZoom =
+            includesZoom ||
+            (inp.hasAttribute('type') &&
+              inp.getAttribute('type').toLowerCase() === 'zoom');
+          if (inp.tagName.toLowerCase() === 'map-select') {
+            // use a throwaway div to parse the input from MapML into HTML
+            var div = document.createElement('div');
+            div.insertAdjacentHTML('afterbegin', inp.outerHTML);
+            // parse
+            inp.htmlselect = div.querySelector('map-select');
+            inp.htmlselect = transcribe(inp.htmlselect);
+
+            // this goes into the layer control, so add a listener
+            L.DomEvent.on(
+              inp.htmlselect,
+              'change',
+              this._layer.redraw,
+              this._layer
+            );
+            if (!this._layer._userInputs) {
+              this._layer._userInputs = [];
+            }
+            this._layer._userInputs.push(inp.htmlselect);
+          }
+          // TODO: if this is an input@type=location
+          // get the TCRS min,max attribute values at the identified zoom level
+          // save this information as properties of the mapExtent,
+          // perhaps as a bounds object so that it can be easily used
+          // later by the layer control to determine when to enable
+          // disable the layer for drawing.
+        } else {
+          console.log(
+            'input with name=' +
+              varName +
+              ' not found for template variable of same name'
+          );
+          // no match found, template won't be used
+          break;
+        }
+      }
+      if (
+        (template && vcount.length === inputs.length) ||
+        template === BLANK_TT_TREF
+      ) {
+        if (trel === 'query') {
+          this._layer.queryable = true;
+        }
+        if (!includesZoom && zoomInput) {
+          inputs.push(zoomInput);
+        }
+        let step = zoomInput ? zoomInput.getAttribute('step') : 1;
+        if (!step || step === '0' || isNaN(step)) step = 1;
+        // template has a matching input for every variable reference {varref}
+        templateVars.push({
+          template: decodeURI(new URL(template, base)),
+          linkEl: t,
+          title: title,
+          rel: trel,
+          type: ttype,
+          values: inputs,
+          zoomBounds: zoomBounds,
+          boundsFallbackPCRS: { bounds: boundsFallback.bounds },
+          projectionMatch: projectionMatch,
+          projection: this.units || FALLBACK_PROJECTION,
+          tms: tms,
+          step: step
+        });
+      }
+    }
+    return templateVars;
+  }
+  createLayerControlExtentHTML() {
```

Comment:
We could probably put this method in its own file, like we did with the similar method for the layer.  Also, if we could make sure that it sets some useful properties on map-extent like the similar layer method did for layer- , we coudl use those properties in tests to simplify the visual complexity of the tests at least.